user/user.go

```diff
@@ -554,12 +555,13 @@ func (u *User) HashPassword(pw, salt string) error {
 	}
 }
 
-func (u *User) PasswordsMatch(pw, salt string) bool {
+func (u *User) PasswordsMatch(pw, salt string, log *log.Logger) bool {
 	if u.PwHash == "" || pw == "" {
 		return false
 	} else if pwMatch, err := GeneratePasswordHash(u.Id, pw, salt); err != nil {
 		return false
 	} else {
+		log.Debugf("user pw hash: %s vs calculated: %s", u.PwHash, pwMatch)
```

Comment:
Not sure it helps the debuging to print the password hash. The important point it to have a log that says the passwords do not match, which is already done in the calling method.
It forces to add the logger in the method signature just for this.